src/metabase/api/dashboard.clj

```diff
@@ -784,38 +786,50 @@
   "How many results to return when chain filtering"
   1000)
 
-(mu/defn ^:private mappings->field-ids :- [:maybe [:set ms/PositiveInt]]
-  [parameter-mappings :- [:maybe [:or [:set dashboard-card/ParamMapping] [:sequential dashboard-card/ParamMapping]]]]
-  (set (for [{{:keys [card]} :dashcard :keys [target]} parameter-mappings
-             :let  [field-clause (params/param-target->field-clause target card)]
-             :when field-clause
-             :let  [{:keys [result_metadata]} card
-                    field-id (or
-                              ;; Get the field id from the field-clause if it contains it. This is the common case for
-                              ;; mbql queries.
-                              (mbql.u/match-one field-clause [:field (id :guard integer?) _] id)
-                              ;; Attempt to get the field clause from the model metadata corresponding to the field.
-                              ;; This is the common case for native queries in which mappings from original columns
-                              ;; have been performed using model metadata.
-                              (:id (qp.util/field->field-info field-clause result_metadata)))]
-             :when field-id]
-         field-id)))
-
-(defn- param-key->field-ids
-  "Get Field ID(s) associated with a parameter in a Dashboard.
-
-    (param-key->field-ids (t2/select-one Dashboard :id 62) \"ee876336\")
-    ;; -> #{276}"
-  [dashboard param-key]
-  {:pre [(string? param-key)]}
-  (let [{:keys [resolved-params]} (t2/hydrate dashboard :resolved-params)
-        param                     (get resolved-params param-key)]
-    (mappings->field-ids (:mappings param))))
-
-(defn- chain-filter-constraints [dashboard constraint-param-key->value]
-  (into {} (for [[param-key value] constraint-param-key->value
-                 field-id          (param-key->field-ids dashboard param-key)]
-             [field-id value])))
+(defn- get-template-tag
```

Comment:
if we don't use template tag for anything, maybe we can makes it returns the :dimension so we don't have to destruct it later on?
```suggestion
(defn- get-template-tag-dimension
```

Reply:
I've spent considerable amount of time yesterday trying to understand why I was getting options from a template tag, if it's not necessary now — and all tests pass. 

Then I began writing this comment and realized that I haven't checked the UI, and voila! — options come from the template tag there and not from the parameter. I'm not quite understanding the difference between tests and real UI here, so need some time to figure it out to write a correct test.

Reply:
can you tell me how to setup a template tag with options?

Reply:
Not programatically yet, but if you set up a dashboard, `param` (incoming in `param->fields`) will look like this:

<details>
<summary>param</summary>
<pre><code>
{:filteringParameters ["fadff97c" "f57f3a34" "864d5df1"],
 :id "38b52427",
 :mappings #{{:card_id 1,
              :dashcard (toucan2.instance/instance
                         :model/DashboardCard
                         {:action_id nil,
                          :card (toucan2.instance/instance
                                 :model/Card
                                 {:archived false,
                                  :cache_ttl nil,
                                  :collection_id nil,
                                  :collection_position nil,
                                  :collection_preview true,
                                  :created_at #t "2023-10-19T08:43:47.374840Z",
                                  :creator_id 1,
                                  :database_id 2,
                                  :dataset false,
                                  :dataset_query {:database 2,
                                                  :native {:query "select * from test\nwhere 1=1\n[[and {{ this_field }} ]]\n[[and {{ exclude_this_field }} ]]\n[[and {{ a_starts_with }} ]]\n[[and {{ amount_between }} ]]",
                                                           :template-tags {"a_starts_with" {:dimension [:field 138 nil],
                                                                                            :display-name "A Starts With",
                                                                                            :id "bb54a514-99e3-4de7-a139-aae7729b7ff9",
                                                                                            :name "a_starts_with",
                                                                                            :options {:case-sensitive false},
                                                                                            :type :dimension,
                                                                                            :widget-type :string/contains},
                                                                           "amount_between" {:dimension [:field 114 nil],
                                                                                             :display-name "Amount Between",
                                                                                             :id "a92931fe-43ec-499e-8072-e34a93f4f260",
                                                                                             :name "amount_between",
                                                                                             :type :dimension,
                                                                                             :widget-type :number/between},
                                                                           "exclude_this_field" {:dimension [:field
                                                                                                             137
                                                                                                             nil],
                                                                                                 :display-name "Exclude This Field",
                                                                                                 :id "7b95074e-25eb-493a-8f0a-22dcc44e09e4",
                                                                                                 :name "exclude_this_field",
                                                                                                 :type :dimension,
                                                                                                 :widget-type :string/!=},
                                                                           "this_field" {:default nil,
                                                                                         :dimension [:field 138 nil],
                                                                                         :display-name "This Field",
                                                                                         :id "7e9d5188-b81f-4f58-8e27-a2a0ff3c681f",
                                                                                         :name "this_field",
                                                                                         :type :dimension,
                                                                                         :widget-type :string/=}}},
                                                  :type :native},
                                  :description nil,
                                  :display :table,
                                  :embedding_params nil,
                                  :enable_embedding false,
                                  :entity_id "sdQYFZgioanVovTzqcAv0",
                                  :id 1,
                                  :made_public_by_id nil,
                                  :metabase_version "v0.48.0-SNAPSHOT (dc676e1)",
                                  :name "ques 34039",
                                  :parameter_mappings [],
                                  :parameters [{:id "7e9d5188-b81f-4f58-8e27-a2a0ff3c681f",
                                                :name "This Field",
                                                :slug "this_field",
                                                :target [:dimension [:template-tag "this_field"]],
                                                :type :string/=}
                                               {:id "7b95074e-25eb-493a-8f0a-22dcc44e09e4",
                                                :name "Exclude This Field",
                                                :slug "exclude_this_field",
                                                :target [:dimension [:template-tag "exclude_this_field"]],
                                                :type :string/!=}
                                               {:id "bb54a514-99e3-4de7-a139-aae7729b7ff9",
                                                :name "A Starts With",
                                                :options {:case-sensitive false},
                                                :slug "a_starts_with",
                                                :target [:dimension [:template-tag "a_starts_with"]],
                                                :type :string/contains}
                                               {:id "a92931fe-43ec-499e-8072-e34a93f4f260",
                                                :name "Amount Between",
                                                :slug "amount_between",
                                                :target [:dimension [:template-tag "amount_between"]],
                                                :type :number/between}],
                                  :public_uuid nil,
                                  :query_type :native,
                                  :result_metadata ({:base_type :type/Integer,
                                                     :display_name "id",
                                                     :effective_type :type/Integer,
                                                     :field_ref [:field "id" {:base-type :type/Integer}],
                                                     :fingerprint nil,
                                                     :name "id",
                                                     :semantic_type :type/PK}
                                                    {:base_type :type/Integer,
                                                     :display_name "amount",
                                                     :effective_type :type/Integer,
                                                     :field_ref [:field "amount" {:base-type :type/Integer}],
                                                     :fingerprint {:global {:distinct-count 2, :nil% 0.0},
                                                                   :type {:type/Number {:avg 2.5,
                                                                                        :max 3.0,
                                                                                        :min 2.0,
                                                                                        :q1 2.0,
                                                                                        :q3 3.0,
                                                                                        :sd 0.7071067811865476}}},
                                                     :name "amount",
                                                     :semantic_type nil}
                                                    {:base_type :type/Boolean,
                                                     :display_name "is_checked",
                                                     :effective_type :type/Boolean,
                                                     :field_ref [:field "is_checked" {:base-type :type/Boolean}],
                                                     :fingerprint {:global {:distinct-count 2, :nil% 0.0}},
                                                     :name "is_checked",
                                                     :semantic_type nil}
                                                    {:base_type :type/Text,
                                                     :display_name "a",
                                                     :effective_type :type/Text,
                                                     :field_ref [:field "a" {:base-type :type/Text}],
                                                     :fingerprint {:global {:distinct-count 2, :nil% 0.0},
                                                                   :type {:type/Text {:average-length 3.0,
                                                                                      :percent-email 0.0,
                                                                                      :percent-json 0.0,
                                                                                      :percent-state 0.0,
                                                                                      :percent-url 0.0}}},
                                                     :name "a",
                                                     :semantic_type nil}
                                                    {:base_type :type/Text,
                                                     :display_name "b",
                                                     :effective_type :type/Text,
                                                     :field_ref [:field "b" {:base-type :type/Text}],
                                                     :fingerprint {:global {:distinct-count 2, :nil% 0.0},
                                                                   :type {:type/Text {:average-length 3.0,
                                                                                      :percent-email 0.0,
                                                                                      :percent-json 0.0,
                                                                                      :percent-state 0.0,
                                                                                      :percent-url 0.0}}},
                                                     :name "b",
                                                     :semantic_type nil}),
                                  :table_id nil,
                                  :updated_at #t "2023-10-31T07:06:49.144983Z",
                                  :visualization_settings {:table.cell_column "amount", :table.pivot_column "id"}}),
                          :card_id 1,
                          :col 0,
                          :collection_authority_level nil,
                          :created_at #t "2023-10-19T08:44:14.822842Z",
                          :dashboard_id 2,
                          :dashboard_tab_id nil,
                          :entity_id "Z95aWfOkx1FZGk0OXVQRN",
                          :id 1,
                          :parameter_mappings [{:card_id 1,
                                                :parameter_id "fadff97c",
                                                :target [:dimension [:template-tag "this_field"]]}
                                               {:card_id 1,
                                                :parameter_id "f57f3a34",
                                                :target [:dimension [:template-tag "exclude_this_field"]]}
                                               {:card_id 1,
                                                :parameter_id "38b52427",
                                                :target [:dimension [:template-tag "a_starts_with"]]}
                                               {:card_id 1,
                                                :parameter_id "864d5df1",
                                                :target [:dimension [:template-tag "amount_between"]]}],
                          :row 0,
                          :size_x 12,
                          :size_y 9,
                          :updated_at #t "2023-10-20T14:08:31.905235Z",
                          :visualization_settings {}}),
              :parameter_id "38b52427",
              :target [:dimension [:template-tag "a_starts_with"]]}},
 :name "Text starts with",
 :sectionId "string",
 :slug "text_starts_with",
 :type :string/starts-with}
</code></pre>
</details>

So no `:options` in param itself, unlike in dashboard_test.clj

Reply:
tests for this are in e0e142abc41482fc2224fc754641a16aa5d1e2d7